mail_gateway/models/mail_gateway.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2024 Dixmit
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import Command, api, fields, models, tools
+
+
+class MailGateway(models.Model):
+    _name = "mail.gateway"
+    _description = "Mail Gateway"
+
+    name = fields.Char(required=True)
+    token = fields.Char(required=True, help="Key used for integration purposes")
+    gateway_type = fields.Selection([], required=True)
+    webhook_key = fields.Char(help="Key used on the connection URL")
+    webhook_secret = fields.Char(
+        help="""Key used to ensure that the connection is secure and
+        comes from the desired source"""
+    )
+    # TODO: Maybe we need to secure this information, isn't it?
```

Comment:
What do you mean?

Reply:
That I need to use the password widget for some fields :wink:

Reply:
In your hands then!